pkg/vdr/web/resolver_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+)
+
+const (
+	prefix = "did:web:"
+
+	validURL         = "www.example.org"
+	validURLWithPath = "www.example.org/user/example"
+	validDID         = prefix + validURL
+	validDIDWithPath = prefix + "www.example.org:user:example"
+
+	invalidDIDNoMethod = "did:" + validURL
+	invalidDIDNoPrefix = validURL
+)
+
+func TestParseDID(t *testing.T) {
+	t.Run("test parse did success", func(t *testing.T) {
+		url, err := parseDIDWeb(validDID)
+		require.NoError(t, err)
+		require.Equal(t, "https://"+validURL+defaultPath, url)
+		url, err = parseDIDWeb(validDIDWithPath)
+		require.NoError(t, err)
+		require.Equal(t, "https://"+validURLWithPath+documentPath, url)
+	})
+
+	t.Run("test parse did failure", func(t *testing.T) {
+		url, err := parseDIDWeb(invalidDIDNoMethod)
+		require.Error(t, err)
+		require.Empty(t, url)
+		require.Contains(t, err.Error(), "does not conform to generic did standard")
+		url, err = parseDIDWeb(invalidDIDNoPrefix)
+		require.Error(t, err)
+		require.Empty(t, url)
+		require.Contains(t, err.Error(), "does not conform to generic did standard")
+	})
+}
+
+func TestResolveDID(t *testing.T) {
+	t.Run("test resolve did with default client failure", func(t *testing.T) {
+		v, err := New()
+		require.NoError(t, err)
+		doc, err := v.Read("did:web:www.example123.org")
+		require.Nil(t, doc)
+		require.Contains(t, err.Error(), "http request with default client unsuccessful")
+	})
+	t.Run("test resolve did with custom client failure", func(t *testing.T) {
+		v, err := New()
+		require.NoError(t, err)
+		doc, err := v.Read("did:web:www.example123.org", vdr.WithHTTPClient(&http.Client{}))
+		require.Nil(t, doc)
+		require.Contains(t, err.Error(), "http request with custom client unsuccessful")
```

Comment:
Please check `require.Error(t, err)` before checking the error message. We avoid a potential panic this way.